Flaky test ExporterDirectorDistributionTest.shouldDistributeExporterPositions

**Summary**

> Try to answer the following as best as possible

- How often does the test fail? I saw it now, I haven't seen it before, but I did not dig into the history to find out.
- Does it block your work? I had to tell bors to merge again
- Do we suspect that it is a real failure? I don't understand the stacktrace. It says that line 97 in the `ExporterDirectorDistributionTest` called `Constructor.newInstanceWithCaller(Constructor.java:499)`, but  [ExporterDirectorDistributionTest.java#L97](https://github.com/camunda-cloud/zeebe/blob/korthout-recreate-benchmark/broker/src/test/java/io/camunda/zeebe/broker/exporter/stream/ExporterDirectorDistributionTest.java#L97)
 is `assertThat(activeExporterState.getPosition(EXPORTER_ID_1)).isEqualTo(position);`

**Logs**

> If possible, provide more context here, e.g. standard output logs, link to build, etc.

<details><summary>Logs</summary>

[link to logs](https://ci.zeebe.camunda.cloud/blue/organizations/jenkins/camunda-cloud%2Fzeebe/detail/staging/1566/pipeline)

<pre>
[2021-12-24T09:27:11.162Z] [ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 4.711 s <<< FAILURE! - in io.camunda.zeebe.broker.exporter.stream.ExporterDirectorDistributionTest
[2021-12-24T09:27:11.162Z] [ERROR] io.camunda.zeebe.broker.exporter.stream.ExporterDirectorDistributionTest.shouldDistributeExporterPositions  Time elapsed: 1.826 s  <<< FAILURE!
[2021-12-24T09:27:11.162Z] org.opentest4j.AssertionFailedError: 
[2021-12-24T09:27:11.162Z] 
[2021-12-24T09:27:11.162Z] expected: 1L
[2021-12-24T09:27:11.162Z]  but was: -1L
[2021-12-24T09:27:11.162Z] 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2021-12-24T09:27:11.162Z] 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
[2021-12-24T09:27:11.162Z] 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2021-12-24T09:27:11.162Z] 	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
[2021-12-24T09:27:11.162Z] 	at io.camunda.zeebe.broker.exporter.stream.ExporterDirectorDistributionTest.shouldDistributeExporterPositions(ExporterDirectorDistributionTest.java:97)
[2021-12-24T09:27:11.162Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2021-12-24T09:27:11.162Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2021-12-24T09:27:11.162Z] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2021-12-24T09:27:11.162Z] 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[2021-12-24T09:27:11.162Z] 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
[2021-12-24T09:27:11.162Z] 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
[2021-12-24T09:27:11.162Z] 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
[2021-12-24T09:27:11.162Z] 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
[2021-12-24T09:27:11.162Z] 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
[2021-12-24T09:27:11.162Z] 	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
[2021-12-24T09:27:11.162Z] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
[2021-12-24T09:27:11.162Z] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
[2021-12-24T09:27:11.162Z] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
[2021-12-24T09:27:11.162Z] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
[2021-12-24T09:27:11.162Z] 	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
[2021-12-24T09:27:11.162Z] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
[2021-12-24T09:27:11.162Z] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
[2021-12-24T09:27:11.162Z] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
[2021-12-24T09:27:11.162Z] 	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
[2021-12-24T09:27:11.162Z] 	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
[2021-12-24T09:27:11.162Z] 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
[2021-12-24T09:27:11.162Z] 	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
[2021-12-24T09:27:11.162Z] 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
[2021-12-24T09:27:11.162Z] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
[2021-12-24T09:27:11.162Z] 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
[2021-12-24T09:27:11.162Z] 	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
[2021-12-24T09:27:11.162Z] 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
[2021-12-24T09:27:11.162Z] 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
[2021-12-24T09:27:11.162Z] 	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
[2021-12-24T09:27:11.162Z] 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
[2021-12-24T09:27:11.162Z] 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
[2021-12-24T09:27:11.162Z] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
[2021-12-24T09:27:11.162Z] 	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
[2021-12-24T09:27:11.162Z] 	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
[2021-12-24T09:27:11.162Z] 	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:42)
[2021-12-24T09:27:11.162Z] 	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:80)
[2021-12-24T09:27:11.162Z] 	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:72)
[2021-12-24T09:27:11.163Z] 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
[2021-12-24T09:27:11.163Z] 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
[2021-12-24T09:27:11.163Z] 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
[2021-12-24T09:27:11.163Z] 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
[2021-12-24T09:27:11.163Z] 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
[2021-12-24T09:27:11.163Z] 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
[2021-12-24T09:27:11.163Z] 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
[2021-12-24T09:27:11.163Z] 	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
[2021-12-24T09:27:11.163Z] 	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)
[2021-12-24T09:27:11.163Z] 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.lambda$execute$1(JUnitPlatformProvider.java:199)
[2021-12-24T09:27:11.163Z] 	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
[2021-12-24T09:27:11.163Z] 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:193)
[2021-12-24T09:27:11.163Z] 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)
[2021-12-24T09:27:11.163Z] 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:120)
[2021-12-24T09:27:11.163Z] 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
[2021-12-24T09:27:11.163Z] 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
[2021-12-24T09:27:11.163Z] 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
[2021-12-24T09:27:11.163Z] 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
</pre>
</details>

Moep I was able to reproduce it via running the complete package in a loop (in intellij)

```
14:32:22.170 [Broker-0-LogStream-1] [-zb-actors-0] DEBUG io.camunda.zeebe.logstreams - Configured log appender back pressure at partition 1 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
14:32:23.682 [Broker-0-LogStream-1] [-zb-actors-2] DEBUG io.camunda.zeebe.logstreams - Configured log appender back pressure at partition 1 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
14:32:25.131 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Recovering exporter from snapshot
14:32:25.352 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Recovered exporter 'exporter' from snapshot at lastExportedPosition -1
14:32:25.352 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Configure exporter with id 'exporter-1'
14:32:25.354 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Configure exporter with id 'exporter-2'
14:32:25.355 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Set event filter for exporters: ExporterEventFilter{acceptRecordTypes={SBE_UNKNOWN=true, COMMAND_REJECTION=true, NULL_VAL=true, EVENT=true, COMMAND=true}, acceptValueTypes={PROCESS=true, NULL_VAL=true, MESSAGE_START_EVENT_SUBSCRIPTION=true, PROCESS_INSTANCE_RESULT=true, DEPLOYMENT=true, VARIABLE_DOCUMENT=true, PROCESS_INSTANCE_CREATION=true, TIMER=true, DEPLOYMENT_DISTRIBUTION=true, MESSAGE=true, PROCESS_EVENT=true, ERROR=true, INCIDENT=true, PROCESS_MESSAGE_SUBSCRIPTION=true, MESSAGE_SUBSCRIPTION=true, JOB_BATCH=true, SBE_UNKNOWN=true, PROCESS_INSTANCE=true, VARIABLE=true, JOB=true}}
14:32:25.367 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Open exporter with id 'exporter-1'
14:32:25.390 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Open exporter with id 'exporter-2'
14:32:27.490 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Recovering exporter from snapshot
14:32:28.401 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Recovered exporter 'exporter' from snapshot at lastExportedPosition -1
14:32:28.401 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Configure exporter with id 'exporter-1'
14:32:28.411 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Configure exporter with id 'exporter-2'
14:32:28.435 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Set event filter for exporters: ExporterEventFilter{acceptRecordTypes={SBE_UNKNOWN=true, COMMAND_REJECTION=true, NULL_VAL=true, EVENT=true, COMMAND=true}, acceptValueTypes={PROCESS=true, NULL_VAL=true, MESSAGE_START_EVENT_SUBSCRIPTION=true, PROCESS_INSTANCE_RESULT=true, DEPLOYMENT=true, VARIABLE_DOCUMENT=true, PROCESS_INSTANCE_CREATION=true, TIMER=true, DEPLOYMENT_DISTRIBUTION=true, MESSAGE=true, PROCESS_EVENT=true, ERROR=true, INCIDENT=true, PROCESS_MESSAGE_SUBSCRIPTION=true, MESSAGE_SUBSCRIPTION=true, JOB_BATCH=true, SBE_UNKNOWN=true, PROCESS_INSTANCE=true, VARIABLE=true, JOB=true}}
14:32:38.846 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Closed exporter director 'exporter'.
14:32:40.095 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Closed exporter director 'exporter'.
14:32:41.754 [Broker-0-LogStream-1] [-zb-actors-1] INFO  io.camunda.zeebe.logstreams - Close appender for log stream stream
14:32:41.754 [stream-write-buffer] [-zb-actors-1] DEBUG io.camunda.zeebe.dispatcher - Dispatcher closed
14:32:41.754 [Broker-0-LogStream-1] [-zb-actors-1] INFO  io.camunda.zeebe.logstreams - On closing logstream stream close 1 readers
14:32:41.755 [] [main] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers'
14:32:41.755 [] [main] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors'
14:32:41.757 [] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers': closed successfully
14:32:41.777 [] [-zb-actors-5] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors': closed successfully
14:32:41.779 [Broker-0-LogStream-1] [-zb-actors-0] INFO  io.camunda.zeebe.logstreams - Close appender for log stream stream
14:32:41.779 [stream-write-buffer] [-zb-actors-0] DEBUG io.camunda.zeebe.dispatcher - Dispatcher closed
14:32:41.779 [Broker-0-LogStream-1] [-zb-actors-0] INFO  io.camunda.zeebe.logstreams - On closing logstream stream close 0 readers
14:32:41.779 [] [main] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers'
14:32:41.779 [] [main] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors'
14:32:41.828 [] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers': closed successfully
14:32:41.848 [] [-zb-actors-5] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors': closed successfully

org.awaitility.core.ConditionTimeoutException: Condition with alias 'Active Director has distributed positions and passive has received it' didn't complete within 10 seconds because assertion condition defined as a lambda expression in io.camunda.zeebe.broker.exporter.stream.ExporterDirectorDistributionTest that uses io.camunda.zeebe.broker.exporter.stream.ExportersState 
expected: 10L
 but was: -1L.
Expected :10L
Actual   :-1L
<Click to see difference>


	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:165)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:119)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:31)
	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:895)
	at org.awaitility.core.ConditionFactory.untilAsserted(ConditionFactory.java:679)
	at io.camunda.zeebe.broker.exporter.stream.ExporterDirectorDistributionTest.shouldNotResetExporterPositionWhenOldPositionReceived(ExporterDirectorDistributionTest.java:136)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:42)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:80)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:72)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:71)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:30)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
Caused by: org.opentest4j.AssertionFailedError: 
expected: 10L
 but was: -1L
	at jdk.internal.reflect.GeneratedConstructorAccessor48.newInstance(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at io.camunda.zeebe.broker.exporter.stream.ExporterDirectorDistributionTest.lambda$shouldNotResetExporterPositionWhenOldPositionReceived$6(ExporterDirectorDistributionTest.java:138)
	at org.awaitility.core.AssertionCondition.lambda$new$0(AssertionCondition.java:53)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:222)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:209)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)

```

Both `ExporterDirectorDistributionTest` failed

```
14:32:42.558 [Broker-0-LogStream-1] [-zb-actors-2] DEBUG io.camunda.zeebe.logstreams - Configured log appender back pressure at partition 1 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
14:32:43.081 [Broker-0-LogStream-1] [-zb-actors-2] DEBUG io.camunda.zeebe.logstreams - Configured log appender back pressure at partition 1 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
14:32:46.377 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Recovering exporter from snapshot
14:32:46.627 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Recovered exporter 'exporter' from snapshot at lastExportedPosition -1
14:32:46.630 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Configure exporter with id 'exporter-1'
14:32:46.631 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Configure exporter with id 'exporter-2'
14:32:46.634 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Set event filter for exporters: ExporterEventFilter{acceptRecordTypes={SBE_UNKNOWN=true, COMMAND_REJECTION=true, NULL_VAL=true, EVENT=true, COMMAND=true}, acceptValueTypes={PROCESS=true, NULL_VAL=true, MESSAGE_START_EVENT_SUBSCRIPTION=true, PROCESS_INSTANCE_RESULT=true, DEPLOYMENT=true, VARIABLE_DOCUMENT=true, PROCESS_INSTANCE_CREATION=true, TIMER=true, DEPLOYMENT_DISTRIBUTION=true, MESSAGE=true, PROCESS_EVENT=true, ERROR=true, INCIDENT=true, PROCESS_MESSAGE_SUBSCRIPTION=true, MESSAGE_SUBSCRIPTION=true, JOB_BATCH=true, SBE_UNKNOWN=true, PROCESS_INSTANCE=true, VARIABLE=true, JOB=true}}
14:32:46.666 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Open exporter with id 'exporter-1'
14:32:46.667 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Open exporter with id 'exporter-2'
14:32:48.491 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Recovering exporter from snapshot
14:32:48.874 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Recovered exporter 'exporter' from snapshot at lastExportedPosition -1
14:32:48.874 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Configure exporter with id 'exporter-1'
14:32:48.874 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Configure exporter with id 'exporter-2'
14:32:48.884 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Set event filter for exporters: ExporterEventFilter{acceptRecordTypes={SBE_UNKNOWN=true, COMMAND_REJECTION=true, NULL_VAL=true, EVENT=true, COMMAND=true}, acceptValueTypes={PROCESS=true, NULL_VAL=true, MESSAGE_START_EVENT_SUBSCRIPTION=true, PROCESS_INSTANCE_RESULT=true, DEPLOYMENT=true, VARIABLE_DOCUMENT=true, PROCESS_INSTANCE_CREATION=true, TIMER=true, DEPLOYMENT_DISTRIBUTION=true, MESSAGE=true, PROCESS_EVENT=true, ERROR=true, INCIDENT=true, PROCESS_MESSAGE_SUBSCRIPTION=true, MESSAGE_SUBSCRIPTION=true, JOB_BATCH=true, SBE_UNKNOWN=true, PROCESS_INSTANCE=true, VARIABLE=true, JOB=true}}
14:32:58.881 [exporter] [-zb-fs-workers-0] DEBUG io.camunda.zeebe.broker.exporter - Closed exporter director 'exporter'.
14:33:01.983 [exporter] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.broker.exporter - Closed exporter director 'exporter'.
14:33:03.820 [Broker-0-LogStream-1] [-zb-actors-0] INFO  io.camunda.zeebe.logstreams - Close appender for log stream stream
14:33:03.820 [stream-write-buffer] [-zb-actors-0] DEBUG io.camunda.zeebe.dispatcher - Dispatcher closed
14:33:03.820 [Broker-0-LogStream-1] [-zb-actors-0] INFO  io.camunda.zeebe.logstreams - On closing logstream stream close 1 readers
14:33:03.821 [] [main] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers'
14:33:03.821 [] [main] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors'
14:33:03.822 [] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers': closed successfully
14:33:03.824 [] [-zb-actors-0] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors': closed successfully
14:33:03.831 [Broker-0-LogStream-1] [-zb-actors-5] INFO  io.camunda.zeebe.logstreams - Close appender for log stream stream
14:33:03.831 [stream-write-buffer] [-zb-actors-5] DEBUG io.camunda.zeebe.dispatcher - Dispatcher closed
14:33:03.831 [Broker-0-LogStream-1] [-zb-actors-5] INFO  io.camunda.zeebe.logstreams - On closing logstream stream close 0 readers
14:33:03.832 [] [main] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers'
14:33:03.832 [] [main] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors'
14:33:03.832 [] [-zb-fs-workers-1] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers': closed successfully
14:33:03.835 [] [-zb-actors-2] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors': closed successfully

org.awaitility.core.ConditionTimeoutException: Condition with alias 'Active Director has distributed positions and passive has received it' didn't complete within 10 seconds because assertion condition defined as a lambda expression in io.camunda.zeebe.broker.exporter.stream.ExporterDirectorDistributionTest that uses io.camunda.zeebe.broker.exporter.stream.ExportersState, io.camunda.zeebe.broker.exporter.stream.ExportersStatelong 
expected: 1L
 but was: -1L.
Expected :1L
Actual   :-1L
<Click to see difference>


	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:165)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:119)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:31)
	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:895)
	at org.awaitility.core.ConditionFactory.untilAsserted(ConditionFactory.java:679)
	at io.camunda.zeebe.broker.exporter.stream.ExporterDirectorDistributionTest.shouldDistributeExporterPositions(ExporterDirectorDistributionTest.java:109)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
	at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:42)
	at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:80)
	at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:72)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:71)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:30)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
Caused by: org.opentest4j.AssertionFailedError: 
expected: 1L
 but was: -1L
	at jdk.internal.reflect.GeneratedConstructorAccessor48.newInstance(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at io.camunda.zeebe.broker.exporter.stream.ExporterDirectorDistributionTest.lambda$shouldDistributeExporterPositions$3(ExporterDirectorDistributionTest.java:111)
	at org.awaitility.core.AssertionCondition.lambda$new$0(AssertionCondition.java:53)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:222)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:209)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)

```

--- a/broker/src/test/java/io/camunda/zeebe/broker/exporter/stream/ExporterDirectorDistributionTest.java
+++ b/broker/src/test/java/io/camunda/zeebe/broker/exporter/stream/ExporterDirectorDistributionTest.java
@@ -15,12 +15,15 @@ import io.camunda.zeebe.broker.exporter.repo.ExporterDescriptor;
 import io.camunda.zeebe.broker.exporter.util.ControlledTestExporter;
 import io.camunda.zeebe.protocol.impl.record.value.deployment.DeploymentRecord;
 import io.camunda.zeebe.protocol.record.intent.DeploymentIntent;
+import io.camunda.zeebe.util.sched.clock.ControlledActorClock;
 import java.time.Duration;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import org.awaitility.Awaitility;
+import org.awaitility.core.ConditionEvaluationListener;
+import org.awaitility.core.EvaluatedCondition;
 import org.junit.After;
 import org.junit.Before;
 import org.junit.Rule;
@@ -106,6 +109,7 @@ public final class ExporterDirectorDistributionTest {
 
     // then
     Awaitility.await("Active Director has distributed positions and passive has received it")
+        .conditionEvaluationListener(new ClockShifter(activeExporters.getClock()))
         .untilAsserted(
             () -> {
               assertThat(passiveExporterState.getPosition(EXPORTER_ID_1)).isEqualTo(position);
@@ -133,6 +137,7 @@ public final class ExporterDirectorDistributionTest {
 
     final var passiveExporterState = passiveExporters.getExportersState();
     Awaitility.await("Active Director has distributed positions and passive has received it")
+        .conditionEvaluationListener(new ClockShifter(activeExporters.getClock()))
         .untilAsserted(
             () -> {
               assertThat(passiveExporterState.getPosition(EXPORTER_ID_1)).isEqualTo(position);
@@ -152,5 +157,26 @@ public final class ExporterDirectorDistributionTest {
 
     // then - the exported position should not go back
     assertThat(passiveExporterState.getPosition(EXPORTER_ID_1)).isEqualTo(position);
+  }
+
+  /**
+   * Shifts the actor clock by the {@link this#DISTRIBUTION_INTERVAL} after an awaitility condition
+   * was evaluated.
+   *
+   * <p>This makes sure that even if we miss one export position event, we distribute the event
+   * later again, which makes tests less flaky.
+   */
+  private static final class ClockShifter implements ConditionEvaluationListener<Void> {
+
+    private final ControlledActorClock clock;
+
+    public ClockShifter(final ControlledActorClock clock) {
+      this.clock = clock;
+    }
+
+    @Override
+    public void conditionEvaluated(final EvaluatedCondition<Void> condition) {
+      clock.addTime(DISTRIBUTION_INTERVAL);
+    }
   }
 }

